chains/solana/contracts/programs/ccip-router/src/instructions/v1/token_admin_registry.rs

```diff
@@ -1,77 +1,165 @@
+use crate::{
+    events::token_admin_registry as events,
+    token_context::{
+        OverridePendingTokenAdminRegistryByCCIPAdmin, OverridePendingTokenAdminRegistryByOwner,
+    },
+};
 use anchor_lang::prelude::*;
 use anchor_spl::associated_token::get_associated_token_address_with_program_id;
 use solana_program::{address_lookup_table::state::AddressLookupTable, log::sol_log};
 
 use super::pools::token_admin_registry_writable;
 
 use crate::{
-    seed, AcceptAdminRoleTokenAdminRegistry, AdministratorRegistered,
-    AdministratorTransferRequested, AdministratorTransferred, CcipRouterError,
-    ModifyTokenAdminRegistry, PoolSet, RegisterTokenAdminRegistryViaGetCCIPAdmin,
-    RegisterTokenAdminRegistryViaOwner, SetPoolTokenAdminRegistry,
+    seed,
+    token_context::{RegisterTokenAdminRegistryByCCIPAdmin, RegisterTokenAdminRegistryByOwner},
+    AcceptAdminRoleTokenAdminRegistry, CcipRouterError, ModifyTokenAdminRegistry,
+    SetPoolTokenAdminRegistry,
 };
 
 const MINIMUM_TOKEN_POOL_ACCOUNTS: usize = 9;
 
-pub fn register_token_admin_registry_via_get_ccip_admin(
-    ctx: Context<RegisterTokenAdminRegistryViaGetCCIPAdmin>,
-    mint: Pubkey, // should we validate that this is a real token program?
+// Proposes an administrator for the given token as pending administrator
```

Comment:
Nit: Would be good to have these be doc comments (/// instead of //) for the `cargo doc` users.